### ISSUE TITLE
[Bug] Cmd4 Crashes Homebridge When Internet Unavailable

### ISSUE THREAD

**Issue by nano9g** [ztalbot2000/homebridge-cmd4#126]
**Describe The Bug:**
When Cmd4 is unable to reach the internet it will put Homebridge into a crash loop.

**To Reproduce:**
Disable internet access for the device hosting Homebridge, then restart Homebridge. Shortly after startup is complete, the log will show the error below and Homebridge will shut down.

**Expected Behaviour:**
Cmd4 should catch the error and log it.

**Paste of Logs:**
```
[2/10/2022, 8:49:44 AM] RequestError: getaddrinfo EAI_AGAIN registry.npmjs.org
    at ClientRequest.<anonymous> (/usr/lib/node_modules/homebridge-cmd4/node_modules/got/source/request-as-event-emitter.js:178:14)
    at Object.onceWrapper (node:events:510:26)
    at ClientRequest.emit (node:events:402:35)
    at ClientRequest.origin.emit (/usr/lib/node_modules/homebridge-cmd4/node_modules/@szmarczak/http-timer/source/index.js:37:11)
    at TLSSocket.socketErrorListener (node:_http_client:447:9)
    at TLSSocket.emit (node:events:390:28)
    at emitErrorNT (node:internal/streams/destroy:157:8)
    at emitErrorCloseNT (node:internal/streams/destroy:122:3)
    at processTicksAndRejections (node:internal/process/task_queues:83:21)
[2/10/2022, 8:49:44 AM] Got SIGTERM, shutting down Homebridge...
```

**Cmd4 Config:**
Not relevant to this bug as it's a core Cmd4 issue.

**Environment:**

* **Node.js Version**: 16.13.2
* **NPM Version**: 8.1.2
* **Homebridge Version**: 1.4.0
* **homebridge-cmd4 Version**: 6.1.1
* **Operating System**: Tested on both Raspbian and Debian
* **Process Supervisor**: hb-service

**Comment by ztalbot2000** [comment]
Hi,

  There are no CMD4 procedures in the callback.  Not unexpected as Cmd4
does not use http or any IP protocols. Homebridge does of course and I
suspect it is their issue.
However, I can dig to try and reproduce it first.
Your complete config.json would be helpful, to see if there are any other
plugins configured. You most likely have Homebridge-ui, for instance.


Ttyl,
John

**Comment by nano9g** [comment]
Thanks for the quick response and sorry for assuming it was definitely a Cmd4 issue. I'll do some more testing and get back to you, hopefully later today.

**Comment by nano9g** [comment]
After additional testing I was able to identify the root cause in the Cmd4 code, so I submitted pull request #127 to resolve the problem. The problem is that there was no error handling for the update check that Cmd4 automatically initiates at startup, so when the `got` library threw an error for the NPM registry hostname being unknown, the error wouldn't be caught and would go all the way up the stack and crash Homebridge.

Now instead of crashing, an error similar to this will be logged:
```
[UPDATE CHECK FAILED] Unable to check for updates to homebridge-cmd4 - RequestError: getaddrinfo EAI_AGAIN registry.npmjs.org
```

FYI, the crash was happening regardless of whether Cmd4 was actually set up in config.json or not - it just needed to be installed and the issue would be triggered when Homebridge started up without internet access. It also didn't matter what other plugins were installed (I stripped my test instance down to only Cmd4 to verify this).

In other words, even with an absolutely minimal config.json like the one below, just having Cmd4 installed was enough to throw Homebridge into a crash loop.

```json
{
    "bridge": {
        "name": "Homebridge Test",
        "username": "##:##:##:##:##:##",
        "port": ####,
        "pin": "###-##-###"
    },
    "platforms": [],
    "accessories": []
}
```

Let me know if you have any questions. Thanks!

**Comment by ztalbot2000** [comment]
Oh Wow!  I love it when users fix things for me.  I wish I could buy you a
coffee!

Thanks ever ever so much.  You do not know how much this means to me!


John
